File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        res = s[0]
        n = len(s)
        for i in range(len(s)):
            j = 1
            while (i-j>=0 and j+i<n):
                if s[i-j]==s[i+j]:
                    if len(res)<2*j+1:
                        res = s[i-j:j+i+1]
                    j+=1
                else:
                    break
            
            #for even
            if i+1<n and (s[i]==s[i+1]):
                if 2>len(res):
                    res = s[i]+s[i+1]
                j=1
                while (i-j>=0 and j+i+1<n):
                    if s[i-j]==s[i+j+1]:
                        if len(res)<2*j+2:
                            res = s[i-j:j+i+2]
                        j+=1
                    else:
                        break
        return res

File: 5-longest-palindromic-substring/longest-palindromic-substring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    
    n = s.length
    let res = s[0]
    for (let i = 0 ; i<n;i++){
        let j = 1
        // consider odd
        while (i-j>=0 && i+j<n){
            if (s[i+j]==s[i-j]){
                if (2*j+1 > res.length){
                    res = s.slice(i-j,i+j+1)
                }
                j+=1
            }
            else{
                break
            }
        } 

        //consider even
        if (i+1<n && s[i]==s[i+1]){ 
            if (2>res.length){
                res = s[i]+s[i+1]
            }
            j=1
            
            while (i-j>=0 && i+j+1<n){
                if (s[i+j+1]==s[i-j]){
                    if (2*j+2 > res.length){
                        res = s.slice(i-j,i+1+j+1)
                    }
                    j+=1
                }
                else{
                    break
                }
            } 
        }
    }
    return res
};